backend/app/assets/javascripts/spree/backend/images/upload.js

```diff
@@ -0,0 +1,204 @@
+      if (!this.tests.formdata) return;
+
+      var progressModel = new ProgressModel({file: file});
+      progressModel.previewFile();
+      progressModel.uploadFile();
+
+      var progressView = new ProgressView({model: progressModel});
+      this.progressZone.appendChild(progressView.render().el);
+    },
+
+    onDragOver: function() {
+      this.el.className = 'hover';
+      return false;
+    },
+
+    onDragLeave: function() {
+      this.el.className = '';
+      return false;
+    },
+
+    onDrop: function(e) {
+      this.el.className = '';
+      e.preventDefault();
+
+      for (var i = 0; i < e.originalEvent.dataTransfer.files.length; i++) {
+        this.upload(e.originalEvent.dataTransfer.files[i]);
+      }
+    },
+
+    onFileBrowserSelect: function(e) {
+      for (var i = 0; i < e.target.files.length; i++) {
+        this.upload(e.target.files[i]);
+      }
+    },
+
+    tests: {
+      filereader: typeof FileReader != 'undefined',
+      dnd: 'draggable' in document.createElement('span'),
+      formdata: !!window.FormData,
+      progress: "upload" in new XMLHttpRequest
+    },
+
+    support: {
+      filereader: document.getElementById('filereader'),
+      formdata:   document.getElementById('formdata'),
+      progress:   document.getElementById('progress')
+    }
+  });
+
+  var ProgressModel = Backbone.Model.extend({
+    initialize: function() {
+      this.set({summary: this.get("file").name});
+    },
+
+    defaults: function() {
+      return {
+        file: null,
+        imgSrc: '',
+        progress: 0,
+        serverError: false,
+        summary: ''
+      }
+    },
+
+    acceptedTypes: {
+      'image/png': true,
+      'image/jpeg': true,
+      'image/gif': true
+    },
+
+    variantId: document.querySelector('input[name="image[viewable_id]"]').value,
+
+    previewFile: function () {
+      var file = this.get('file'),
+          that = this;
+
+      if (FileReader && this.acceptedTypes[file.type] === true) {
+        var reader = new FileReader();
+        reader.onload = function (event) {
+          that.set({imgSrc: event.target.result});
+        };
+
+        reader.readAsDataURL(file);
+      } else {
+        var summary = 'Uploading ' + file.name + ' ' + (file.size ? (file.size/1024|0) + 'K' : '');
+        this.set({summary: summary});
+      }
+    },
+
+    uploadFile: function () {
+      var formData = new FormData(),
+          that = this;
+
+      formData.append('image[attachment]', this.get('file'));
+      formData.append('image[viewable_id]', this.variantId);
+      formData.append('upload_id', this.cid);
+
+      // send the image to the server
+      Spree.ajax({
+        url: window.location.pathname,
+        type: "POST",
+        dataType: 'script',
+        data: formData,
+        processData: false,  // tell jQuery not to process the data
+        contentType: false,  // tell jQuery not to set contentType
+        xhr: function () {
```

Comment:
I think it's too early at this point for anything such. As far as I understand, one only gets progress for large user submitted requests, i.e. files. For regular post requests when the server is just slow .. that's a different story that would like need server support.